### PR TITLE
added progress bar functionality and styling

### PR DESCRIPTION
**Pull request by mwazir**
PR to merge progress bar into main

### CODE REVIEW

**Review by itsninam — APPROVED** [review]

File: src/App.js [hunk 1]
```diff
@@ -8,6 +8,8 @@ import Footer from "./Components/Footer.js";
 import Game from "./Components/Game.js";
 import SlideOutMenu from "./Components/SlideOutMenu.js";
 import LandingPage from "./Components/LandingPage.js";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
```
Comment (benbourgon) [inline src/App.js:11]: do we have a comment to separate components from modules?

**Review by alextfriedman — APPROVED** [review]
amazing work 🥧

File: src/Components/Game.js [hunk 2]
```diff
@@ -100,23 +110,51 @@ const Game = (props) => {
...
       setIsGameOver(prevState => {
         return true;
-      });      
+      });   
     }
   }
 
+  const handleIconUpdate = () => {
+    progressBarIconArray.map((item, index) => {
+      if (index <= questionNumber) {
```
Comment (benbourgon) [inline src/Components/Game.js:130]: Very cool to tie it to the question number as the source of truth

File: src/styles/sass/utilities/_variables.scss [hunk 3]
```diff
@@ -15,6 +15,8 @@ $clr-black: rgba(0, 0, 0, 1);
 $clr-purple: rgba(96, 86, 249, 1);
 $clr-limeGreen: rgba(207, 255, 49, 1);
 $clr-lightBlue: rgba(227, 244, 243,1);
+$clr-red: rgba(214, 69, 65, 1);
```
Comment (benbourgon) [inline src/styles/sass/utilities/_variables.scss:18]: Might need to change the colour here for contrast

**Review by benbourgon — APPROVED** [review]
Incredible stuff!! 😵⚰️

**Pull request merged**